vgscollect-nfc/src/main/java/com/verygoodsecurity/api/nfc/core/ReadTagRunnable.kt

```diff
@@ -1,42 +1,78 @@
 package com.verygoodsecurity.api.nfc.core
 
 import android.nfc.Tag
-import com.verygoodsecurity.api.nfc.core.content.CardAdapter
-import com.verygoodsecurity.api.nfc.core.content.CommandAPDU
-import com.verygoodsecurity.api.nfc.core.content.CommandEnum
-import com.verygoodsecurity.api.nfc.core.content.IsoDepProvider
+import com.verygoodsecurity.api.nfc.core.content.*
 import com.verygoodsecurity.api.nfc.core.model.Card
-import com.verygoodsecurity.api.nfc.core.utils.isSucceed
+import com.verygoodsecurity.api.nfc.core.utils.*
 
 internal class ReadTagRunnable(
     @Suppress("unused") private val tag: Tag,
     private val listener: ResultListener,
 ) : Runnable {
 
     private val provider = IsoDepProvider(tag)
-    private val cardAdapter = CardAdapter()
-
 
     override fun run() {
         provider.connect()
 
-        val command = CommandAPDU(CommandEnum.SELECT, PPSE).toBytes()
+        provider.transceive(
+            CommandAPDU(CommandEnum.SELECT, PPSE).toBytes()
+        )?.takeIf { it.isSuccessful() }
+            ?.run {
+                parseFCITemplate(this)
+            }
+            ?.takeIf { it.isSuccessful() }
```

Comment:
@Endorf Not something important, but I would like to use the same pattern for all lines, what I mean, 
?.run {  parseFCITemplate(this) }
?.takeIf { it.isSuccessful() }
?.run { extractCardData(this) }
?.also { listener.onSuccess(it) } ?: listener.onFailure("error")